apps/token-manager/app/src/components/Panels/AssignVotePanelContent.js

```diff
@@ -1,8 +1,11 @@
 import React from 'react'
 import styled from 'styled-components'
 import { Button, Field, IconCross, Text, TextInput } from '@aragon/ui'
-import BN from 'bn.js'
 import { addressPattern, isAddress } from '../../web3-utils'
+import { fromDecimals, toDecimals } from '../../utils'
+
+// Any more and the number input field starts to put numbers in scientific notation
+const MAX_INPUT_DECIMAL_BASE = 6
```

Comment:
Any more and we get this:

<img src="https://user-images.githubusercontent.com/4166642/46916862-bc9ee300-cfc0-11e8-9e83-800c00b72c7d.png" height=150>

Reply:
Do you know if it’s anywhere in the HTML spec, or if it depends on the browser implementation?

We should probably have our own validator in the future, to allow the exact decimal base we want (when we will start working on reusable form validation and errors).

Reply:
From https://www.w3.org/TR/html51/sec-forms.html#number-state-typenumber:

> If the user agent provides a user interface for selecting a number, then the value must be set to the best representation of the number representing the user’s selection as a floating-point number.

> The algorithm to convert a number to a string, given a number input, is as follows: Return a valid floating-point number that represents input.

> The best representation of the number n as a floating-point number is the string obtained from running ToString(n). The abstract operation ToString is not uniquely determined. When there are multiple possible strings that could be obtained from ToString for a particular value, the user agent must always return the same string for that value (though it may differ from the value used by other user agents).

Seems like it's pretty up in the air.

Reply:
@bpierre Can confirm, firefox implements this differently. They don't quite support all the way up to 18 decimals, but pretty close.

<img src="https://user-images.githubusercontent.com/4166642/47022489-e6414100-d15d-11e8-9cab-9b2050b73abc.png" height=150>